scripts/trim_indel_ends.py

```diff
@@ -0,0 +1,402 @@
+        - Trim prefix and suffix only when the rest of the alignment is all matches --- set a threshold
+"""
+
+import pandas as pd
+from collections import defaultdict, Counter
+import re
+import edlib
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+from Bio.Seq import Seq
+import matplotlib.pyplot as plt
+from tqdm import tqdm
+import os
+import sys
+import argparse
+import datetime
+
+#--------------------------------------------------------
+# parser
+#--------------------------------------------------------
+# positional/optional arguments
+posList = ['manifest','unique_isoform_aln']
+optList = ['mode', 'threshold']
+sys.stdout.write('\n'); sys.stdout.flush()
+
+# author and version info
+usage = sys.argv[0] + ' <options> ' + ' '.join(posList) + '\n'
+author = 'Author: Yutong Qiu (yutongq@andrew.cmu.edu)'
+version = 'Version: 1.0.0.0 (2022-11-23)'
+description = '\nDescription: This program trims isoform sequences to reduce false positives. Output trimmed sequences in trimmed.fasta'
+
+parser = argparse.ArgumentParser(usage = '\n'.join([usage, author, version, description]), formatter_class=argparse.RawTextHelpFormatter)
+# positional arguments
+parser.add_argument(posList[0], type=str, help='string\tinput manifest csv file in format: sample,fa.  e.g. /in/manifest.csv')
+parser.add_argument(posList[1], type=str, help='string\talignment between similar isoforms within each window.  e.g. /in/unique_isoforms_and_aln_stats.tsv')
+# optional arguments
+parser.add_argument('-m', '--'+optList[0], type=str, metavar='', default='lenient', help='string\tMode of trimming. "Strict" only trims when sequences match exactly after trimming; "Lenient" trims whenever prefix and suffix indel numbers below threshold, defauls to "Lenient"')
+parser.add_argument('-t', '--'+optList[1], type=int, metavar='', default=10000, help='int\tThreshold for trimming, defaults to 10000 (trim every prefix and suffix)')
+
+# strict positional/optional arguments checking
+argsDict = vars(parser.parse_args())
+sys.stdout.write('\n' + datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - ' + sys.argv[0] + ' - Parsing Input Arguements...\n\n'); sys.stdout.flush()
+for key, value in argsDict.items():
+    if key in posList: sys.stdout.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - Required Argument - ' + key +': '+ str(value) + '\n'); sys.stdout.flush()
+    if key in optList: sys.stdout.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - Optional Argument - ' + key +': '+ str(value) + '\n'); sys.stdout.flush()
+    vars()[key] = value # assign values of arguments into shorthand global variables
+sys.stdout.write('\n'); sys.stdout.flush()
+
+def extract_edits(cigar_str) -> list:
+    """Given a cigar string, return separated edits with counts
+
+    Args:
+        cigar_str (str):
+
+    Returns:
+        list: a list of tuples of (count, edit_type)
```

Comment:
@ame9yu Could you give an example of the expected output here? That would speed up unit tests and potential debugging. 

I think there's a standard way to do this with python docs (CC @cmatKhan )

e.g.

```
""""
Explanation

Args: 

Returns:

Example:

"""
```

Reply:
I added some examples in the new commit

Reply:
This is good for unit tests, thank you 

CC @cmatKhan